src/app/shared/components/template/components/task-card/task-card.component.ts

```diff
@@ -1,16 +1,91 @@
-import { getStringParamFromTemplateRow } from "src/app/shared/utils";
+import {
+  getStringParamFromTemplateRow,
+  getVariantParamFromTemplateRow,
+} from "src/app/shared/utils";
 import { TemplateBaseComponent } from "../base";
 import { IProgressStatus } from "src/app/shared/services/task/task.service";
 import { TemplateFieldService } from "../../services/template-field.service";
 
+interface ITaskCardParams {
+  /**
+   * The ID of an individual task. A task card can display *either* an individual task,
+   * *or* a task group: either a task_id or a task_group_id should be provided.
+   */
+  task_id: string;
+  /**
+   * The ID of a task group, used to track whether it is the highlighted task group.
+   * If provided, the task card will display a progress bar for the task group
+   * **/
+  task_group_id: string;
+  /**
+   * The name of the data_list containing the elements of the task group,
+   * used to track completion progress of the task group
+   * **/
+  task_group_data: string;
+  /**
+   * The name of the field that tracks the completion of the task. Can be omitted when displaying a task group,
+   * as their completion is calculated by the completion of their constituent tasks
+   */
+  completed_field: string;
+  /** The title to display on the task card */
+  title: string;
+  /**
+   * The subtitle to display on the task card, typically a short description.
+   * Optional
+   */
+  subtitle: string;
+  /**
+   * The image to display on the task card.
+   * Optional
+   */
+  image: string;
+  /** Legacy, use variant instead */
+  style: string;
+  /**
+   * A list of named style variants of the component, separated by spaces or commas.
+   * Use interchangeably with "variant_list". Default "landscape"
+   * */
+  variant: string;
+  /**
+   * A list of named style variants of the component, separated by spaces or commas.
+   * Use interchangeably with "variant". Default "landscape"
+   * */
+  variant_list: string;
+  /**
+   * The icon to display in the "badge" added to the task card
+   * when its associated task/task group has been completed
+   */
+  completed_icon: string;
+  /**
+   * The icon to display in the "badge" added to the task card
+   * when its associated task/task group is in progress
+   */
+  in_progress_icon: string;
+  /**
+   * Text to display in the "badge" added to the task card
+   * when its associated task group is the highlighted task group.
+   * Default "Active"
+   */
+  highlighted_text: string;
+  /**
+   * The displayed name of the units of progress associated with
+   * the constituent tasks of the task group.
+   * Default "sections"
+   */
+  progress_units_name: string;
+}
+
 @Component({
   selector: "plh-task-card",
   templateUrl: "./task-card.component.html",
   styleUrls: ["./task-card.component.scss"],
 })
 export class TmplTaskCardComponent extends TemplateBaseComponent implements OnInit {
+  /** Authoring parameters */
+  public parameter_list: ITaskCardParams;
```

Comment:
This isn't actually used, but included here in order to link to the type in the generated docs. Alternatively to keeping the separate properties, we could store all the relevant properties in a `parameter_list` object and read them off, as in `radio-button-grid.component.ts`, however this removes the use of `getStringParamFromTemplateRow()` etc, which doesn't seem desirable. Eventually, perhaps we would want to handle the `get<type>ParamFromTemplateRow()` in the parser itself and remove the reliance on these utility functions, which would allow neater organisation of a component's properties within a `parameter_list` object.